Src/FluentAssertions/Specialized/ActionAssertions.cs

```diff
@@ -35,6 +38,8 @@ public ActionAssertions(Action subject, IExtractExceptions extractor)
         public ExceptionAssertions<TException> Throw<TException>(string because = "", params object[] becauseArgs)
             where TException : Exception
         {
+            FailIfSubjectAsyncVoid("Throw");
```

Comment:
Let me know if you disagree with adding the check on Should().Throw() as well. It doesn't need it as much as NotThrow IMO, but could help users avoid confusion while writing tests.

Reply:
Maybe use `nameof`

Reply:
afaik nameof doesn't let you capture the correct (generic or non-generic) NotThrow method. Not that I expect one name to change and not the other, but still seemed wrong.